File: ZooParade/Assets/ZooParade/Scripts/Map.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class Map
{
	public string name;
	public Animal[] animals;
	public Checkpoint[] checkpoints;
	public Player player;

	public Map(string name, Animal[] animals, Checkpoint[] checkpoints)
	{
		this.name = name;
		this.animals = animals;
		this.checkpoints = checkpoints;
	}

	public void AssignPlayer(Player player)
	{
		this.player = player;
	}

	public List <Checkpoint> GetNextCheckpoint ()
	{
		return player.currentCheckpoint.GetNext ();
	}

	public void MovePlayer(Checkpoint checkpoint)
	/*
	 * Moves the player assigned to this map to the specified checkpoint.
	 */
	{
		// Move the Player to the next Checkpoint using MovePlayer()
		player.MovePlayer (checkpoint);
	}


}

File: map.js
/* Initialzer for a Map object
 * Parameter types : (string, list of Animal, list of Checkpoint)
 */
function Map(name, animals, checkpoints){
	this.name = name;
	this.animals = animals;
	this.checkpoints = checkpoints;
	this.player = null;
}


/* Get all possible moves steps 'steps' away from checkpoint and 
 * returns an array of checkpoints
 * Parameter types : (Map)
 */
function GetPossibleMoves(map){
	// To be implemented
}


/* Assigns player to map
 * Parameter types : (Map, Player)
 */
function AssignPlayer(map, player){
	map.player = player;
}